test/integration/integration_admin_test.cc

```diff
@@ -442,4 +444,78 @@ TEST_F(IntegrationAdminIpv4Ipv6Test, Ipv4Ipv6Listen) {
   }
 }
 
+// Testing the behavior of StatsMatcher, which allows/denies the  instantiation of stats based on
+// restrictions on their names.
+class StatsMatcherIntegrationTest : public HttpIntegrationTest, public testing::Test {
+public:
+  StatsMatcherIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, Network::Address::IpVersion::v4,
+                            simTime()) {}
+
+  void initialize() override {
+    config_helper_.addConfigModifier(
+        [this](envoy::config::bootstrap::v2::Bootstrap& bootstrap) -> void {
+          *bootstrap.mutable_stats_config()->mutable_stats_matcher() = stats_matcher_;
+        });
+    HttpIntegrationTest::initialize();
+
+    response_ = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/stats", "",
```

Comment:
to me it's not intuitive that you are initiating a request from a method called initialize(). Seems like that might belong in the body of various tests.

As an alternative, can you have a separate method to make the admin request and check its status (removing the checking from TearDown())?

Reply:
Will separate these out into better-named functions.